core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -181,6 +182,7 @@ class GroupCoordinator(val brokerId: Int,
           responseCallback(JoinGroupResult(memberId, Errors.UNKNOWN_MEMBER_ID))
         case Some(group) =>
           group.inLock {
+            info(s"memberId=$memberId with groupInstanceId=$groupInstanceId is attempting to join groupId=$groupId due to: $reason")
```

Comment:
Since the consumer might be in the old version, the `reason` might be `null`. I think we should not output the log when the `reason` is null. i.e.:
`memberId=1 with groupInstanceId=2 is attempting to join groupId=3 due to: null`